docs/elements.md

```diff
@@ -1244,6 +1244,56 @@ As of right now, the element supports displaying either Pandas DataFrames as an
 - [`pl-code` to display blocks of code with syntax highlighting](#pl-code-element)
 - [`pl-variable-output` for displaying a matrix or element in code form.](#pl-variable-output-element)
 
+### `pl-dataframe` element
+
+Displays a formatted display of Pandas DataFrames, with various options for displaying types of columns and code for reproducing the DataFrame.
+
+#### Sample elements
+
+**question.html**
+
+```html
+<pl-dataframe params-name="df" show-index="false" show-dimensions="false" digits="4"></pl-dataframe>
+```
+
+**server.py**
+
+```python
+import prairielearn as pl
+import pandas as pd
+
+def generate(data):
+  df = pd.io.parsers.read_csv("breast-cancer-train.dat", header=None)
+  data["params"]["df"] = pl.to_json(df.head(15))
+```
+
+#### Customizations
+
+| Attribute          | Type          | Default  | Description                                                                                                                                            |
+| ------------------ | ------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `params-name`      | string        | —        | The name of the key in `data['params']` to get a value from.                                                                                           |
+| `show-header`      | boolean       | true     | Show the header row of a DataFrame.                                                                                                                    |
+| `show-index`       | boolean       | true     | Show the index column of a DataFrame.                                                                                                                  |
+| `show-dimensions`  | boolean       | true     | Show a footer with the dimensions of a DataFrame.                                                                                                      |
+| `show-dtype`       | boolean       | false    | Show the data types contained in each column of the DataFrame at the bottom of each column. Types used correspond to the `display-language` parameter. |
+| `display-language` | "python", "r" | "python" | Language to use for displaying data types and indices.                                                                                                 |
+| `show-python`      | boolean       | true     | Show code that can be used to recreate the DataFrame in Python in a separate tab.                                                                      |
+| `digits`           | integer       | -        | Number of digits to display for floating point entries.                                                                                                |
+
+#### Details
+
+When setting a parameter, use PrairieLearn's built in `pl.to_json()` on the DataFrame to display.
```

Comment:
Should there be a separate documentation page somewhere that describes `pl.to_json()` and `pl.from_json()` serialization/unserialization efforts? 

We probably should mention here that `pl.to_json(df, df_encoding_version=2)` allows for missing and date time values whereas `pl.to_json(df, df_encoding_version=1)` (default) does not. However, it has support for complex numbers.

Reply:
Docs added!

Reply:
FWIW, I filed https://github.com/PrairieLearn/PrairieLearn/issues/6965 to actually put these into the user-facing docs.